src/logic/errors.js

```diff
@@ -0,0 +1,77 @@
+import {createLogic} from 'redux-logic';
+import Analyzer from '../analyzers';
+import {getCurrentProject} from '../selectors';
+import {validatedSource} from '../actions/errors';
+import retryingFailedImports from '../util/retryingFailedImports';
+
+export async function importValidations() {
+  return retryingFailedImports(() =>
+    import(
+      /* webpackChunkName: "mainAsync" */
+      '../validations'
+    ),
+  );
+}
+
+const validateSource = async (
+  {language, source, projectAttributes},
+  dispatch,
+) => {
+  const validations = await importValidations();
+  const validate = validations[language];
+  const validationErrors = await validate(source, projectAttributes);
+  await dispatch(validatedSource(language, validationErrors));
+};
```

Comment:
Rather than putting multiple logics in one module, let’s move this into its own module, probably inside a `src/logic/helpers` package that we can use in the general case when we need to share code between different logics. Then we can put each of the logics defined here in its own module.